### PR TITLE
feature: adds sphinx documentation and RTD infra

### PR DESCRIPTION
**Pull request by tallpress**
This is how the docs will appear https://eva-python-sdk-testing.readthedocs.io/en/feature-documentation/evasdk.html url will change
Changes
- adds Spinx dep. to create documentation. Installs Sphinx dep to generate module documentation https://www.sphinx-doc.org/en/master/
- adds dep. m2r2 to Sphinx include `README.md` in the docs https://crossnox.github.io/m2r2/
- adds .readthedocs.yaml, which should mean that RTD can read and build our docs with Sphinx.
- removes badges from README.md heading as it was messing with the table of contents
- adds RTD automation step that will automatically build semver tagged versions of the SDK's documentation

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -32,3 +32,20 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+      - name: Install dependencies
+        run: pip install Sphinx ; pip install m2r2
+      - name: Update the docs
+        run: cd docs ; sphinx-apidoc -f  -o . ..
+      - name: Commit changes
+        uses: EndBug/add-and-commit@v7
```
Comment (LouisBrunner) [inline .github/workflows/build.yml]: I am not convinced by that

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -32,3 +32,20 @@ jobs:
...
         token: ${{secrets.CODECOV_TOKEN}}
         file: ./coverage.xml
+
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+      - name: Install dependencies
+        run: pip install Sphinx ; pip install m2r2
```
Comment (LouisBrunner) [inline .github/workflows/build.yml]: Use pipenv?

File: README.md [hunk 3]
```diff
@@ -174,6 +177,37 @@ $ pipenv run test --ip 172.16.16.2 --token abc-123-def-456
...
+
+We have a *GitHub Action* in the `build.yaml` workflow that will do this automatically update the documentation and commit the documentation changes on pushing a branch to GitHub. For this we are using [EndBug/add-and-commit](https://github.com/EndBug/add-and-commit).
+
+*Read The Docs* has [automation rules](https://docs.readthedocs.io/en/stable/automation-rules.html).
+We have one set up that will automatically build documentation for any semver tagged version of the project.
+
+When adding new files in `evasdk/`, it's important to remember to update the documentation by running the following:
+```bash
+cd docs
+sphinx-apidoc -f  -o . ..
```
Comment (LouisBrunner) [inline README.md]: Add it to the Makefile?

**Review by LouisBrunner — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (tallpress) [reply .github/workflows/build.yml]: how come?

File: .github/workflows/build.yml (hunk 2 above)
Comment (tallpress) [reply .github/workflows/build.yml]: sure, can use the current Pipfile. My only concern is maybe it's a little overkill to create a virtualenv on the unbuntu instance just to install sphinx and m2r2?

**Comment by tallpress** [comment]
closing in favour of https://github.com/automata-tech/eva_python_sdk/pull/89

**Pull request closed without merging**